packages/game-passes/contracts/GamePasses.sol

```diff
@@ -0,0 +1,1161 @@
+     * @dev Verifies the signature, checks supply limits, processes payment, and mints tokens
+     * @dev Updates the per-wallet minting count and transfers payment to the appropriate treasury
+     * @dev Reverts if:
+     *      - Contract is paused
+     *      - Token is not configured
+     *      - Signature is invalid or expired
+     *      - Max supply would be exceeded
+     *      - Max per wallet would be exceeded
+     *      - Payment transfer fails
+     */
+    function mint(
+        address from,
+        uint256 tokenId,
+        uint256 amount,
+        uint256 price,
+        uint256 deadline,
+        bytes memory signature
+    ) external whenNotPaused {
+        address caller = _msgSender();
+
+        if (from != caller) {
+            revert InvalidSender(from, caller);
+        }
+
+        TokenConfig storage config = tokenConfigs[tokenId];
+
+        if (!config.isConfigured) {
+            revert TokenNotConfigured(tokenId);
+        }
+        MintRequest memory request = MintRequest({
+            caller: caller,
+            tokenId: tokenId,
+            amount: amount,
+            price: price,
+            deadline: deadline,
+            nonce: nonces[caller]++
+        });
+
+        verifySignature(request, signature);
+        _checkMaxPerWallet(tokenId, caller, amount);
+        _checkMaxSupply(tokenId, amount);
+
+        // Update minted amount for wallet
+        config.mintedPerWallet[caller] += amount;
+
+        // Transfer payment
+        address treasury = config.treasuryWallet != address(0) ? config.treasuryWallet : defaultTreasuryWallet;
+        SafeERC20.safeTransferFrom(IERC20(paymentToken), _msgSender(), treasury, price);
+
+        // Mint tokens
+        _mint(caller, tokenId, amount, "");
+    }
+
+    /**
+     * @notice Batch mints multiple tokens with valid EIP-712 signatures in a single transaction
+     * @param from Address that will receive the tokens (must be same as msg.sender)
+     * @param tokenIds Array of token IDs to mint
+     * @param amounts Array of amounts to mint for each token ID
+     * @param prices Array of prices to pay for each mint operation
+     * @param deadlines Array of timestamps after which each signature becomes invalid
+     * @param signatures Array of EIP-712 signatures from authorized signers
+     * @dev Processes multiple mint operations in batch, verifying each signature
+     * @dev All array parameters must be the same length
+     * @dev Updates per-wallet minting counts and transfers payments to appropriate treasuries
+     * @dev Reverts if:
+     *      - Contract is paused
+     *      - Array lengths don't match
+     *      - Any token is not configured
+     *      - Any signature is invalid or expired
+     *      - Any max supply would be exceeded
+     *      - Any max per wallet would be exceeded
+     *      - Any payment transfer fails
+     */
+    function batchMint(
```

Comment:
There is no upper limit on `batchMint` and there should be to avoid issues with uncapped array length

Reply:
Added a batch limit of a 100 in https://github.com/thesandboxgame/sandbox-smart-contracts/pull/1619/commits/4315ce1755437e9ed961fec68e444654c005a691

The amount to be confirmed.